wallet/walletrpc.c

```diff
@@ -69,6 +70,35 @@ static void wallet_withdrawal_broadcast(struct bitcoind *bitcoind,
 	}
 }
 
+/**
+ * segwit_addr_net_decode - Try to decode a Bech32 address and detect
+ * testnet/mainnet
+ *
+ * This processes the address and returns true if it is a Bech32
+ * address specified by BIP173. If it returns true, then *testnet is
+ * set whether it is testnet "tb" address or false if mainnet "bc"
+ * address. It does not check, witness version and program size
+ * restrictions.
+ */
+static bool segwit_addr_net_decode(bool *testnet, int *witness_version,
```

Comment:
We'll eventually want to support more than just bitcoin testnet and mainnet, so we may want to wire this up with `chainparams.c` and return an integer/enum indicating the type or an error. But for now this seems reasonable.